### PR TITLE
add new metrics for CDN invalidation queue & execution time

### PR DESCRIPTION
**Pull request by syphar**

### CODE REVIEW

File: src/metrics/mod.rs [hunk 1]
```diff
@@ -29,6 +29,10 @@ metrics! {
 
         /// queued CDN invalidations
         queued_cdn_invalidations_by_distribution: IntGaugeVec["distribution"],
+        /// duration of CDN invalidations after having beeing sent to the CDN.
```
Comment (jsha) [inline src/metrics/mod.rs:32]:
```suggestion
        /// duration of CDN invalidations after having been sent to the CDN.
```

**Review by jsha — COMMENTED** [review]
What are the default buckets for the HistogramVec? We should make aure they're appropriate.

In particular any bucket less than 60s is going to be misleading because it depends on the timing of the repeating task.

**Comment by syphar** [comment]
> What are the default buckets for the HistogramVec? We should make aure they're appropriate.
> 
> In particular any bucket less than 60s is going to be misleading because it depends on the timing of the repeating task.

Valid point, I didn't (don't?) know enough about prometheus yet. 

A first look at the code _looks_ like there are [`DEFAULT_BUCKETS`](https://docs.rs/prometheus/latest/src/prometheus/histogram.rs.html#25-27) that are probably used: 
```rust
/// The default [`Histogram`] buckets. The default buckets are
/// tailored to broadly measure the response time (in seconds) of a
/// network service. Most likely, however, you will be required to define
/// buckets customized to your use case.
pub const DEFAULT_BUCKETS: &[f64; 11] = &[
    0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 1.0, 2.5, 5.0, 10.0,
];
```

Do you know more about buckets and how they are defined? So we can choose better ones? 

Otherwise I would take the time to dig into it the next days.

**Comment by jsha** [comment]
Normally you define the buckets when you create the metric. For us, creation of the metric is slightly hidden in our load_metric_type! macro defined locally, so that will require a little tweaking. We need to figure out how to [pass one of these](https://docs.rs/prometheus/latest/prometheus/struct.HistogramOpts.html) where right now we say `        pub(crate) cdn_queue_time: HistogramVec["distribution"],`.

**Comment by syphar** [comment]
I already got that far, the gap in my knowledge is more around how exactly histogram buckets work, so which would be the buckets that we have to use for this metric

**Comment by syphar** [comment]
( I already found a source, will come back with a proposal :) )

**Comment by syphar** [comment]
@jsha I added more suitable buckets (I think). 

This _feels_ like it's shoved into the current macro design, and would need some refactoring, but I would postpone that to antoher PR.

File: src/metrics/macros.rs [hunk 2]
```diff
@@ -38,6 +40,24 @@ macro_rules! metrics {
                 Ok(Self {
                     registry,
                     recently_accessed_releases: RecentlyAccessedReleases::new(),
+                    cdn_invalidation_time: prometheus::HistogramVec::new(
+                        prometheus::HistogramOpts::new(
+                            "cdn_invalidation_time",
+                            "duration of CDN invalidations after having beeing sent to the CDN.",
```
Comment (jsha) [inline src/metrics/macros.rs:46]:
```suggestion
                            "duration of CDN invalidations after having been sent to the CDN.",
```

**Review by jsha — APPROVED** [review]

**Pull request merged**